source/agent/webrtc/wrtcConnection.js

```diff
@@ -138,56 +232,65 @@ class WrtcStream {
 }
 
 
-module.exports = function (spec, on_status, on_mediaUpdate) {
-  var that = {},
-    direction = spec.direction,
-    wrtcId = spec.connectionId,
-    threadPool = spec.threadPool,
-    ioThreadPool = spec.ioThreadPool,
-    // preferredAudioCodecs = spec.preferred_audio_codecs,
-    // preferredVideoCodecs = spec.preferred_video_codecs,
-    networkInterfaces = spec.network_interfaces,
-    audio = !!spec.media.audio,
-    video = !!spec.media.video,
-    formatPreference = spec.formatPreference,
-    audio_fmt,
-    video_fmt,
-    audioFrameConstructor,
-    audioFramePacketizer,
-    videoFrameConstructor,
-    videoFramePacketizer,
-    simulcastConstructors = [],
-    stream,
-    simStreams = [],
-    ridStreamMap = new Map(),
-    isSimulcast = false,
-    transportSeqNumExt = -1,
-    wrtc;
+module.exports = function (spec, on_status, on_track) {
+  var that = {};
+  var wrtcId = spec.connectionId;
+  var wrtc = null;
+  var threadPool = spec.threadPool;
+  var ioThreadPool = spec.ioThreadPool;
+  var networkInterfaces = spec.network_interfaces;
+
+  var remoteSdp = null;
+  var localSdp = null;
+  // mid => { operationId, type, sdpDirection, formatPreference, rids, enabled, finalFormat }
+  var operationMap = new Map();
```

Comment:
An operation is a transceiver?

Reply:
is a publication or subscription, multiple transceivers